File: C++/0020. Exponentiation.cpp
int main(){
    string s1;
    int n;
    while(cin>>s1>>n){
        size_t id = s1.find('.');
        int deci = (s1.size()-1-id)*n;
        s1 = s1.substr(0,id) + s1.substr(id+1);
        
        int l1 = s1.size();
        string s2 = s1;
        while(n>1){
            int l2 = s2.size();
            string res(l1+l2,0);
            for(int i=l1-1;i>=0;--i){
                for(int j=l2-1;j>=0;--j){
                    res[i+j+1] += (s1[i]-'0')*(s2[j]-'0');
                    res[i+j] += res[i+j+1]/10;
                    res[i+j+1] %= 10;
                }
            }
            
            for(auto &c:res){
                c += '0';
            }
            
            int i = 0;
            while(res[i]=='0'){
                i += 1;
            }
            s2 = res.substr(i);
            n--;
        }
        if(deci > s2.size()){
            s2 = string(deci-s2.size(),'0')+s2;
        }
        s2 = s2.substr(0,s2.size()-deci)+"."+s2.substr(s2.size()-deci);
        
        int i = s2.size()-1;
        while(i>s2.size()-deci and s2[i]=='0'){
            i--;
        }
        s2 = s2.substr(0,i+1);
        
        cout<<s2<<endl;
    }
    
}

File: Python/0020. Exponentiation.py
# If you need to import additional packages or classes, please import here.

def func():
    while True:
        try:
            s2,n = input().strip().split()
            
            idx = s2.find('.')
            s2 = s2[:idx]+s2[idx+1:]
            n = int(n)
            deci = n*idx
      
            s1 = s2
            l1 = len(s1)
            while n>1:
                l2 = len(s2)
                res = [0]*(l1+l2)
                for i in range(l1-1,-1,-1):
                    for j in range(l2-1,-1,-1):
                        res[i+j+1] += int(s1[i])*int(s2[j])
                        res[i+j] += res[i+j+1]//10
                        res[i+j+1] %= 10
                s2 = ("").join(str(c) for c in res)
                n -= 1
            l = len(s2)
            if l < deci:
                s2 = '0'*(deci-len(s2)) + s2
            s2 = s2[:deci] + '.' + s2[deci:]
            
            i = 0
            while s2[i] == '0':
                i += 1
            s2 = s2[i:]
            
            i = len(s2)-1
            while s2[i] == '0':
                i -= 1
            s2 = s2[:i+1]
            
            print(s2)
        except EOFError:
            break
    
if __name__ == "__main__":
    func()
